### ISSUE TITLE
PermissionDenied when setting Execution Policy

### ISSUE THREAD

**Issue by motoridersd** [AntoineGS/TeamsStatusV2#9]
It appears my company has locked down my work computer with a policy that overrides the Execution Policy. Running the installer results in an error

```
Set-ExecutionPolicy : Windows PowerShell updated your execution policy successfully, but the setting is overridden by
a policy defined at a more specific scope.  Due to the override, your shell will retain its current effective
execution policy of Unrestricted. Type "Get-ExecutionPolicy -List" to view your execution policy settings. For more
information please see "Get-Help Set-ExecutionPolicy".
At C:\Scripts\Install.ps1:4 char:1
+ Set-ExecutionPolicy -ExecutionPolicy RemoteSigned
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : PermissionDenied: (:) [Set-ExecutionPolicy], SecurityException
    + FullyQualifiedErrorId : ExecutionPolicyOverride,Microsoft.PowerShell.Commands.SetExecutionPolicyCommand
```
Probably not much that can be done, but I didn't see this come up in other issues on this or the older project. 

After that error, the installation script throws other errors. Password was entered correctly. Username and password have been scrubbed from output below

```
Service "Microsoft Teams Status Monitor" removed successfully!Service "Microsoft Teams Status Monitor" installed successfully!
Please enter the password for user username: 
Failed to look up the SID for username .\userame!
LsaLookupNames(): No mapping between account names and security IDs was done.

Failed to look up the SID for username .\username!
LsaLookupNames(): No mapping between account names and security IDs was done.

Failed to grant the "Log on as a service" right to account .\username!
Error setting parameter "ObjectName" for service "Microsoft Teams Status Monitor"!Start-Service : Failed to start service 'Microsoft Teams Status Monitor (Microsoft Teams Status Monitor)'.At C:\Scripts\Install.ps1:28 char:1+ Start-Service -Name "Microsoft Teams Status Monitor"+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~    + CategoryInfo          : OpenError: (System.ServiceProcess.ServiceController:ServiceController) [Start-Service],   ServiceCommandException    + FullyQualifiedErrorId : StartServiceFailed,Microsoft.PowerShell.Commands.StartServiceCommand
```

**Comment by AntoineGS** [comment]
You might be able to get around the security if you are able to execute the main script "Get-TeamsStatus.ps1" directly.
If that works then you could try replicating the steps from the installer manually, and running nssm.exe to set it up as a service.
But I have a feeling installing services will also not work if your machine has been locked down by your company :(

**Comment by Kuglikrug** [comment]
I have the same issue. Running the main script "Get-TeamsStatus.ps1" directly works fine.
How can I set up my laptop to run the "Get-TeamsStatus.ps1" script automatically in the background without the "Install.ps1"?

Like @motoridersd already said, i did not have the permissions issue when running the V1 Version:

```
Set-ExecutionPolicy -ExecutionPolicy RemoteSigned
Unblock-File .\Settings.ps1
Unblock-File .\Get-TeamsStatus.ps1
Start-Process -FilePath .\nssm.exe -ArgumentList 'install "Microsoft Teams Status Monitor" "C:\Windows\System32\WindowsPowerShell\v1.0\powershell.exe" "-command "& { . C:\Scripts\Get-TeamsStatus.ps1 }"" ' -NoNewWindow -Wait
Start-Service -Name "Microsoft Teams Status Monitor"
```

https://github.com/EBOOZ/TeamsStatus

**Comment by buhito81** [comment]
> After that error, the installation script throws other errors. Password was entered correctly. Username and password have been scrubbed from output below
> 
> ```
> Service "Microsoft Teams Status Monitor" removed successfully!Service "Microsoft Teams Status Monitor" installed successfully!
> Please enter the password for user username: 
> Failed to look up the SID for username .\userame!
> LsaLookupNames(): No mapping between account names and security IDs was done.
> 
> Failed to look up the SID for username .\username!
> LsaLookupNames(): No mapping between account names and security IDs was done.
> 
> Failed to grant the "Log on as a service" right to account .\username!
> Error setting parameter "ObjectName" for service "Microsoft Teams Status Monitor"!Start-Service : Failed to start service 'Microsoft Teams Status Monitor (Microsoft Teams Status Monitor)'.At C:\Scripts\Install.ps1:28 char:1+ Start-Service -Name "Microsoft Teams Status Monitor"+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~    + CategoryInfo          : OpenError: (System.ServiceProcess.ServiceController:ServiceController) [Start-Service],   ServiceCommandException    + FullyQualifiedErrorId : StartServiceFailed,Microsoft.PowerShell.Commands.StartServiceCommand
> ```

This is one of the issues I had too. My work device is added to the AzureAD domain, which it looks like the script doesn't take into account. I've manually changed line 26 in Install.ps1, and that resolved that specific error.

From:
`$ArgumentList = 'set "Microsoft Teams Status Monitor" ObjectName .\' + "$env:USERNAME $Password"`

To:
`$ArgumentList = 'set "Microsoft Teams Status Monitor" ObjectName AzureAD\' + "$env:USERNAME $Password"`

(I'm not skilled sufficiently to alter the PowerShell Scripts. but I think that the script should add the environment variables for USERDOMAIN in case there is a userdomain.)

Still, I cannot add it as a service (see https://github.com/AntoineGS/TeamsStatusV2/issues/13), but if I run Get-TeamsStatus.ps1 the Teams Status is updated in HA.

**Comment by AntoineGS** [comment]
Hmm, I do not have a Windows 11 machine to test but I can definitely add a variable for userdomain!
I likely won't get to it this week but I'll make some time next week for this!
It makes sense that it would work with the EBOOZ version as I believe it would install it as a SYSTEM service instead of being under the user (which caused some issues).

**Comment by buhito81** [comment]
I'm happy to help, and on workdays usually have a Win11 machine at hand.

**Comment by AntoineGS** [comment]
Hey, a lot has been done recently that could affect this (including moving away from nssm and the need to install a service).
While the new method is a little less elegant, it does seem to be much less prone to installation issues.
Can you check if your issue has been addressed with v1.1.0?
Thanks!
